From a4f510c4f630893ed698f8d50540d3efaf69e5c6 Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Sun, 27 Nov 2011 16:08:38 +0000
Subject: Disable failing GNU/Hurd tests ext/Socket/t/socketpair.t

These tests currently hang on GNU/Hurd, so disable them

Bug-Debian: http://bugs.debian.org/650186

Patch-Name: debian/hurd_test_skip_socketpair.diff
---
 ext/Socket/t/socketpair.t |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/ext/Socket/t/socketpair.t b/ext/Socket/t/socketpair.t
index 997628c..24447b1 100644
--- a/ext/Socket/t/socketpair.t
+++ b/ext/Socket/t/socketpair.t
@@ -8,7 +8,8 @@ BEGIN {
     require Config; import Config;
     $can_fork = $Config{'d_fork'} || $Config{'d_pseudofork'};
 
-    if ($^O eq "hpux" or $Config{'extensions'} !~ /\bSocket\b/ &&
+    # fails on GNU/Hurd (Debian #650186)
+    if ($^O eq "gnu" or $^O eq "hpux" or $Config{'extensions'} !~ /\bSocket\b/ &&
         !(($^O eq 'VMS') && $Config{d_socket})) {
 	print "1..0\n";
 	exit 0;
